### PR TITLE
change AUTO default hint to Latency

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *change AUTO default hint to Latency*

### Tickets:
 - *ticket-97729*

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -329,18 +329,22 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
...
     loadConfig.UpdateFromMap(config, GetName(), true);
     auto fullConfig = loadConfig._keyConfigMap;
+    bool workModeAuto = GetName() == "AUTO";
     // Remove the performance hint if no setting to this property from user.
     if (!loadConfig._isSetPerHint) {
         fullConfig.erase(PluginConfigParams::KEY_PERFORMANCE_HINT);
-        // set performance hint to 'THROUGHPUT' model for AutoExecutable Network.
         loadConfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT, PluginConfigParams::THROUGHPUT);
+        if (workModeAuto) {
+            // set performance hint to 'THROUGHPUT' model for AutoExecutable Network.
```
Comment (yangwang201911) [inline src/plugins/auto/plugin.cpp:338]:
```suggestion
            // set performance hint to 'LATENCY' model for AutoExecutable Network.
```

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp [hunk 2]
```diff
@@ -1110,16 +1110,16 @@ TEST_P(OVClassLoadNetworkWithDefaultPropertiesTest, LoadNetworkWithDefaultProper
...
-    ASSERT_EQ(value, ov::hint::PerformanceMode::THROUGHPUT);
+    ASSERT_EQ(value, ov::hint::PerformanceMode::LATENCY);
 }
 
 TEST_P(OVClassLoadNetworkWithDefaultIncorrectPropertiesTest, LoadNetworkWithDefaultIncorrectPropertiesTest) {
     ov::Core ie = createCoreWithTemplate();
     ov::CompiledModel model;
     OV_ASSERT_NO_THROW(model = ie.compile_model(actualNetwork, target_device, configuration));
-    ov::hint::PerformanceMode value = ov::hint::PerformanceMode::THROUGHPUT;
+    ov::hint::PerformanceMode value = ov::hint::PerformanceMode::LATENCY;
```
Comment (wangleis) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:1120]: Should keep THROUGHPUT test case for MULTI. Create new test case for AUTO with default LATENCY.

**Review by wangleis — CHANGES_REQUESTED** [review]

File: src/plugins/auto/plugin.cpp [hunk 3]
```diff
@@ -329,18 +329,22 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     // updateFromMap will check config valid
     loadConfig.UpdateFromMap(config, GetName(), true);
     auto fullConfig = loadConfig._keyConfigMap;
+    bool workModeAuto = GetName() == "AUTO";
     // Remove the performance hint if no setting to this property from user.
     if (!loadConfig._isSetPerHint) {
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:334]: Need to check if Hint is set in config

File: src/plugins/auto/plugin.cpp [hunk 4]
```diff
@@ -329,18 +329,22 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
...
         fullConfig.erase(PluginConfigParams::KEY_PERFORMANCE_HINT);
-        // set performance hint to 'THROUGHPUT' model for AutoExecutable Network.
         loadConfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT, PluginConfigParams::THROUGHPUT);
+        if (workModeAuto) {
```
Comment (wangleis) [inline src/plugins/auto/plugin.cpp:337]: please also check if user set performance hint during compile_model()

**Review by wangleis — CHANGES_REQUESTED** [review]

File: src/plugins/auto/plugin.cpp [hunk 5]
```diff
@@ -329,18 +329,22 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
...
+            // set performance hint to 'LATENCY' model for AutoExecutable Network.
+            loadConfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT,
+                                                  PluginConfigParams::LATENCY);
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:340]: if set perf hint here, the updated logic in setDefaultHint will never be hit, @yangwang201911 , need your double check

File: src/plugins/auto/plugin.cpp (hunk 4 above)
Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:337]: The above code line 330 has been checked if performance hint is set.
`loadConfig.UpdateFromMap(config, GetName(), true);`

File: src/plugins/auto/plugin.cpp (hunk 3 above)
Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:334]: The above code at line 330 has been checked if performance hint is set.
`loadConfig.UpdateFromMap(config, GetName(), true);`

File: src/plugins/auto/plugin.cpp (hunk 5 above)
Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:340]: setting perf hint here just update the perf hint within the Auto executable network. The logic of `setDefaultHint` will work as expected.

Comment (songbell) [reply src/plugins/auto/plugin.cpp:340]: then can line 336 and 339 be optimized?

**Review by yangwang201911 — APPROVED** [review]

File: src/plugins/auto/plugin.cpp (hunk 5 above)
Comment (wgzintel) [reply src/plugins/auto/plugin.cpp:339]: Done

File: src/plugins/auto/plugin.cpp [hunk 6]
```diff
@@ -64,14 +64,14 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
...
     std::vector<std::string> devicesWithRequests = _pluginConfig.ParsePrioritiesDevices(priorities);
 
-    auto setTputAsDefault = [&](const std::string& targetDevice,
+    auto setDefaultHint = [&](const std::string& targetDevice,
                                 std::map<std::string, std::string>& deviceConfig,
                                 const std::map<std::string, std::string>& mergedConfig) {
         auto isSetPerHint = mergedConfig.find(PluginConfigParams::KEY_PERFORMANCE_HINT) != mergedConfig.end();
         if (GetName() == "AUTO" && !isSetPerHint && mergedConfig.find(targetDevice) == mergedConfig.end()) {
-            // setting tput as the default performance mode if no hints setting for AUTO plugin and no properties
+            // setting latency as the default performance mode if no hints setting for AUTO plugin and no properties
```
Comment (wangleis) [inline src/plugins/auto/plugin.cpp:72]: Please check if user set secondary properties like nstreams to HW device. 

If user set those secondary properties, please do not set default hint to HW device.

**Review by wangleis — CHANGES_REQUESTED** [review]

File: src/plugins/auto/plugin.cpp [hunk 7]
```diff
@@ -64,28 +64,33 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
...
-                }
+            auto isSetAffinity = mergedConfig.find(ov::affinity.name()) != mergedConfig.end();
+            auto isSetNumThreads = mergedConfig.find(ov::inference_num_threads.name()) != mergedConfig.end();
+            if (!isSetPerHint && !isSetAffinity && !isSetNumThreads && !isSetDeviceProperties && !isSetNumStreams) {
+                // setting tput as the default performance mode if
+                // 1. no hints setting for MULTI plugin
+                // 2. no affinity setting for MULTI plugin
+                // 3. no inference_num_threads setting for MULTI plugin
+                // 4. no ov::device::properties(secondary properties) setting for target device
+                // 5. no ov::num_streams setting for target device
```
Comment (wangleis) [inline src/plugins/auto/plugin.cpp:91]: Please check num_streams in mergedConfig because MULTI plugin support num_streams also.

**Review by wangleis — CHANGES_REQUESTED** [review]

File: src/plugins/auto/plugin.cpp [hunk 8]
```diff
@@ -64,28 +64,33 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
...
-                if (kvp.first == ov::affinity || kvp.first == ov::num_streams ||
-                    kvp.first == ov::inference_num_threads) {
-                    return;
-                }
+            auto isSetAffinity = mergedConfig.find(ov::affinity.name()) != mergedConfig.end();
+            auto isSetNumThreads = mergedConfig.find(ov::inference_num_threads.name()) != mergedConfig.end();
+            if (!isSetPerHint && !isSetAffinity && !isSetNumThreads && !isSetDeviceProperties && !isSetNumStreams) {
+                // setting tput as the default performance mode if
+                // 1. no hints setting for MULTI plugin
+                // 2. no affinity setting for MULTI plugin
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/auto/plugin.cpp:88]: Please follow affinity part in CVS-98245

**Review by wangleis — APPROVED** [review]

**Pull request merged**